sdk/formrecognizer/azure-ai-formrecognizer/tests/test_invoice_async.py

```diff
@@ -0,0 +1,336 @@
+        self.assertEqual(invoice.fields.get("CustomerName").value, "Microsoft")
+        self.assertEqual(invoice.fields.get("InvoiceId").value, '34278587')
+        self.assertEqual(invoice.fields.get("InvoiceDate").value, date(2017, 6, 18))
+        self.assertEqual(invoice.fields.get("InvoiceTotal").value, 56651.49)
+        self.assertEqual(invoice.fields.get("DueDate").value, date(2017, 6, 24))
+
+    @GlobalFormRecognizerAccountPreparer()
+    @GlobalClientPreparer()
+    async def test_invoice_pdf_include_field_elements(self, client):
+        with open(self.invoice_pdf, "rb") as fd:
+            invoice = fd.read()
+
+        async with client:
+            poller = await client.begin_recognize_invoices(invoice, include_field_elements=True)
+
+            result = await poller.result()
+        self.assertEqual(len(result), 1)
+        invoice = result[0]
+
+        self.assertFormPagesHasValues(invoice.pages)
+
+        for field in invoice.fields.values():
+            self.assertFieldElementsHasValues(field.value_data.field_elements, invoice.page_range.first_page_number)
+
+    @GlobalFormRecognizerAccountPreparer()
+    @GlobalClientPreparer()
+    @pytest.mark.live_test_only
+    async def test_invoice_continuation_token(self, client):
+
+        with open(self.invoice_tiff, "rb") as fd:
+            invoice = fd.read()
+        async with client:
+            initial_poller = await client.begin_recognize_invoices(invoice)
+            cont_token = initial_poller.continuation_token()
+            poller = await client.begin_recognize_invoices(invoice, continuation_token=cont_token)
+            result = await poller.result()
+            self.assertIsNotNone(result)
+            await initial_poller.wait()  # necessary so azure-devtools doesn't throw assertion error
+
+    @GlobalFormRecognizerAccountPreparer()
+    @GlobalClientPreparer(client_kwargs={"api_version": FormRecognizerApiVersion.V2_0})
+    async def test_invoice_v2(self, client):
+        with open(self.invoice_pdf, "rb") as fd:
+            invoice = fd.read()
+        with pytest.raises(ValueError) as e:
+            async with client:
+                await client.begin_recognize_invoices(invoice)
+        assert "Method 'begin_recognize_invoices' is only available for API version V2_1_PREVIEW and up" in str(e.value)
+
+    @GlobalFormRecognizerAccountPreparer()
+    @GlobalClientPreparer()
+    async def test_invoice_locale_specified(self, client):
+        with open(self.invoice_tiff, "rb") as fd:
+            invoice = fd.read()
+        async with client:
+            poller = await client.begin_recognize_invoices(invoice, locale="en-US")
+            assert 'en-US' == poller._polling_method._initial_response.http_response.request.query['locale']
+            await poller.wait()
+
+    @GlobalFormRecognizerAccountPreparer()
+    @GlobalClientPreparer()
+    async def test_invoice_locale_error(self, client):
```

Comment:
would it be good to add a test with multipage invoice?

Reply:
yep, there's an issue for it https://github.com/Azure/azure-sdk-for-python/issues/14402 (won't be in this PR)